src/expr/src/scalar/func.rs

```diff
@@ -4210,6 +4214,33 @@ fn trim_whitespace<'a>(a: Datum<'a>) -> Datum<'a> {
     Datum::from(a.unwrap_str().trim_matches(' '))
 }
 
+fn right<'a>(a: Datum<'a>, b: Datum<'a>) -> Datum<'a> {
+    let string: &'a str = a.unwrap_str();
+    let n = b.unwrap_int64();
+
+    // the index of every char in bytes
+    let byte_indices = string.char_indices().map(|(i, _)| i);
+
+    let start_in_bytes = if n > 0 {
+        // pop off the last n-1 elements
+        let mut byte_indices = byte_indices.rev().skip((n - 1) as usize);
+
+        // this will be our byte index
+        // if we went to far we just take the whole string
+        byte_indices.next().unwrap_or(0)
+    } else {
+        let n = -n;
```

Comment:
Minor, but this could still be negative for `i64::MIN`. The `as usize` later on will do something, probably not as bad as UB, but did want to flag!

Reply:
I would use `i64::abs()`, which leaves it at `i64::MIN`. Or I could convert to `i128` before to make sure it always works, but it seems overkill. WDYT?

Reply:
Not too worried, just ideally avoiding crashing or anything like that.